clients/src/test/java/org/apache/kafka/clients/admin/internals/AdminFetchMetricsManagerTest.java

```diff
@@ -0,0 +1,173 @@
+
+        mockSleepTimeWindow();
+        adminFetchMetricsManager.recordLatency(connectionId0, 11);
+        adminFetchMetricsManager.recordLatency(connectionId1, 111);
+        assertEquals(8, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(11, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(108, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertEquals(11, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(11, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(111, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg1)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax1)));
+
+        adminFetchMetricsManager.recordLatency(connectionId0, 500);
+        adminFetchMetricsManager.recordLatency(connectionId0, 600);
+        mockSleepTimeWindow();
+        adminFetchMetricsManager.recordLatency(connectionId1, 800);
+        adminFetchMetricsManager.recordLatency(connectionId1, 900);
+        assertEquals(550, metricValue(nodeLatencyAvg0), EPSILON);
+        assertEquals(600, metricValue(nodeLatencyMax0), EPSILON);
+        assertEquals(850, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(900, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg0)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax0)));
+        assertEquals(850, metricValue(nodeLatencyAvg1), EPSILON);
+        assertEquals(900, metricValue(nodeLatencyMax1), EPSILON);
+
+        mockSleepTimeWindow();
+        assertTrue(Double.isNaN(metricValue(nodeLatencyAvg1)));
+        assertTrue(Double.isNaN(metricValue(nodeLatencyMax1)));
+    }
+
+    private Map<String, String> genericTag(String connectionId) {
+        Map<String, String> tags = new LinkedHashMap<>();
+        tags.put("node-id", "node-" + connectionId);
+        return tags;
+    }
+
+    private void mockSleepTimeWindow() {
+        time.sleep(metrics.config().timeWindowMs() + 1);
+    }
+
+    private void registerNodeLatencyMetric(String connectionId, MetricName nodeLatencyAvg, MetricName nodeLatencyMax) {
+        String nodeTimeName = "node-" + connectionId + ".latency";
+        Sensor nodeRequestTime = metrics.sensor(nodeTimeName);
+        nodeRequestTime.add(nodeLatencyAvg, new Avg());
+        nodeRequestTime.add(nodeLatencyMax, new Max());
+    }
+
+    private double metricValue(MetricName metricName) {
+        KafkaMetric metric = metrics.metric(metricName);
+        return (Double) metric.metricValue();
```

Comment:
It's a bit odd that you use a primitive type in the method signature but cast the return value to a boxed type. It's better that we could use primitive type here. 
Please fix it.